src/components/TodoContent/TodoContent.tsx

```diff
@@ -0,0 +1,121 @@
+import React, {
+  useMemo,
+  useCallback,
+  useState,
+  useContext,
+} from 'react';
+
+import { TodosContext } from '../../context/todosContext';
+
+import { TodoHeader } from '../TodoHeader';
+import { TodoBody } from '../TodoBody';
+import { TodoFooter } from '../TodoFooter';
+
+import { FilterStatus } from '../../types/FilterStatus';
+
+type Props = {
+  newTodoField: React.RefObject<HTMLInputElement>;
+  onChangeTitle: (title: string) => void;
+  todoTitle: string;
+  createNewTodo: (title: string) => void;
+  isLoading: boolean;
+  deleteTodo: (todoId: number) => void;
+  proccessedTodoIds: number[];
+  changeCompleteStatus: (todoId: number, isComplited: boolean) => void;
+  deleteAllCompletedTodos: () => void;
+  changeTodoTitle: (todoId: number, newTitle: string) => void;
+};
+
+export const TodoContent: React.FC<Props> = React.memo(
+  ({
+    newTodoField,
+    onChangeTitle,
+    todoTitle,
+    createNewTodo,
+    isLoading,
+    deleteTodo,
+    proccessedTodoIds,
+    changeCompleteStatus,
+    deleteAllCompletedTodos,
+    changeTodoTitle,
+  }) => {
+    const [filterStatus, setFilterStatus] = useState(FilterStatus.All);
+    const todos = useContext(TodosContext);
+
+    const countOfTodos = todos.length;
+
+    const isEveryTodoComplited = useMemo(
+      () => todos.every((todo) => todo.completed),
+      [todos],
+    );
+
+    const copleteAllTodos = useCallback(() => {
+      todos.forEach((todo) => {
+        if (!todo.completed) {
+          changeCompleteStatus(todo.id, todo.completed);
+        }
+
+        if (isEveryTodoComplited) {
+          changeCompleteStatus(todo.id, todo.completed);
+        }
+      });
+    }, [todos]);
+
+    const filterTodos = (filterBy: FilterStatus) => {
+      setFilterStatus(filterBy);
+
+      return todos.filter((todo) => {
+        switch (filterBy) {
+          case FilterStatus.Active:
+            return !todo.completed;
+
+          case FilterStatus.Completed:
+            return todo.completed;
+
+          default:
+            return todo;
+        }
+      });
+    };
+
+    const filteredTodos = useMemo(
+      () => filterTodos(filterStatus),
+      [filterStatus, todos],
+    );
```

Comment:
Just filter array and save result to constant using `useMemo`, why to create function for this?